src/es/cuevana/src/eu/kanade/tachiyomi/animeextension/es/cuevana/CuevanaEu.kt

```diff
@@ -0,0 +1,369 @@
+                val urlRequest = "https://$mainUrl/details.php?v=$token"
+                val response = client.newCall(GET(urlRequest, headers = headers)).execute().asJsoup()
+                val bodyText = response.select("body").text()
+                val json = json.decodeFromString<JsonObject>(bodyText)
+                val status = json["status"]!!.jsonPrimitive!!.content
+                val file = json["file"]!!.jsonPrimitive!!.content
+                if (status == "200") { videoList.add(Video(file, "$prefix Tomatomatela", file, headers = null)) }
+            } catch (_: Exception) { }
+        }
+        if (embedUrl.contains("yourupload")) {
+            val videos = YourUploadExtractor(client).videoFromUrl(url, headers = headers)
+            videoList.addAll(videos)
+        }
+        if (embedUrl.contains("doodstream") || embedUrl.contains("dood.")) {
+            DoodExtractor(client).videoFromUrl(url, "$prefix DoodStream", false)
+                ?.let { videoList.add(it) }
+        }
+        if (embedUrl.contains("okru") || embedUrl.contains("ok.ru")) {
+            OkruExtractor(client).videosFromUrl(url, prefix, true).also(videoList::addAll)
+        }
+        if (embedUrl.contains("voe")) {
+            VoeExtractor(client).videoFromUrl(url, "$prefix Voex")?.let { videoList.add(it) }
+        }
+        if (embedUrl.contains("streamtape")) {
+            StreamTapeExtractor(client).videoFromUrl(url, "$prefix StreamTape")?.let { videoList.add(it) }
+        }
+        if (embedUrl.contains("wishembed") || embedUrl.contains("streamwish") || embedUrl.contains("wish")) {
+            StreamWishExtractor(client, headers).videosFromUrl(url, "$prefix StreamWish:").also(videoList::addAll)
+        }
+        if (embedUrl.contains("filemoon") || embedUrl.contains("moonplayer")) {
+            FilemoonExtractor(client).videosFromUrl(url, "$prefix Filemoon:").also(videoList::addAll)
+        }
+        return videoList
+    }
+
+    override fun videoListSelector() = throw Exception("not used")
+
+    override fun videoUrlParse(document: Document) = throw Exception("not used")
+
+    override fun videoFromElement(element: Element) = throw Exception("not used")
+
+    override fun List<Video>.sort(): List<Video> {
+        return try {
+            val videoSorted = this.sortedWith(
+                compareBy<Video> { it.quality.replace("[0-9]".toRegex(), "") }.thenByDescending { getNumberFromString(it.quality) },
+            ).toTypedArray()
+            val userPreferredQuality = preferences.getString("preferred_quality", "Voex")
+            val preferredIdx = videoSorted.indexOfFirst { x -> x.quality == userPreferredQuality }
+            if (preferredIdx != -1) {
+                videoSorted.drop(preferredIdx + 1)
+                videoSorted[0] = videoSorted[preferredIdx]
+            }
+            videoSorted.toList()
+        } catch (e: Exception) {
+            this
+        }
+    }
```

Comment:
This doesn't work as expected, see: https://pl.kotl.in/scJEzSCVX

Reply:
I changed the current method to the method that the other extensions use to keep the same functionality.